### PR TITLE
store uncles information

### PR DESCRIPTION
**Pull request by bailantaotao**

### CODE REVIEW

File: migration/db/migrate/20180710052024_store_uncles_information.rb [hunk 1]
```diff
@@ -0,0 +1,30 @@
+class StoreUnclesInformation < ActiveRecord::Migration[5.2]
+  def up
```
Comment (markya0616) [inline migration/db/migrate/20180710052024_store_uncles_information.rb]: Use update?

File: client/client.go [hunk 2]
```diff
@@ -117,6 +122,30 @@ func (c *client) TransactionReceipts(ctx context.Context, txs types.Transactions
 	return receipts, nil
 }
 
+func (c *client) UncleByBlockHashAndPosition(ctx context.Context, hash common.Hash, position uint) (*types.Header, error) {
+	var result *types.Header
+	err := c.rpc.CallContext(ctx, &result, "eth_getUncleByBlockHashAndIndex", hash, hexutil.Uint(position))
+	if err == nil && result == nil {
```
Comment (markya0616) [inline client/client.go]: Can you also add this check in other client api?

File: client/client_test.go [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/getamis/sirius/log"
+)
+
+func ExampleUncleByNumberAndPosition() {
```
Comment (markya0616) [inline client/client_test.go]: Why do we need the example check?

File: store/store.go [hunk 4]
```diff
@@ -228,16 +230,24 @@ func (m *manager) insertBlock(ctx context.Context, dbTx *gorm.DB, block *types.B
...
 
 	// Insert blocks
-	minerBaseReward, uncleInclusionReward, _, unclesReward, unclesHash := common.AccumulateRewards(block.Header(), block.Uncles())
+	minerBaseReward, uncleInclusionReward, unclesReward, unclesHash := common.AccumulateRewards(block.Header(), block.Uncles())
 	h, err := common.Header(block).AddReward(totalTxsFee, minerBaseReward, uncleInclusionReward, unclesReward, unclesHash)
 	if err != nil {
 		return err
 	}
-	headerStore.Insert(h)
+	err = headerStore.Insert(h)
```
Comment (markya0616) [inline store/store.go]: :ttt:

File: client/client_test.go (hunk 3 above)
Comment (bailantaotao) [reply client/client_test.go]: Just introduce how to use the API and we can debug with this quickly.

Comment (bailantaotao) [reply client/client_test.go]: The [`Output`](https://github.com/getamis/eth-indexer/pull/45/files#diff-fce84029d346504e44323c30d21a3808R27) have double commented `////`, so the example will never be executed.

File: client/client.go (hunk 2 above)
Comment (bailantaotao) [reply client/client.go]: OK

File: migration/db/schema.rb [hunk 5]
```diff
@@ -144,4 +144,26 @@
     t.index ["hash"], name: "index_transactions_on_hash", unique: true
   end
 
+  create_table "uncle_headers", options: "ENGINE=InnoDB DEFAULT CHARSET=utf8 COLLATE=utf8_unicode_ci", force: :cascade do |t|
```
Comment (kimmylin) [inline migration/db/schema.rb]: this table should also index on block_number and be deleted during reorg.
but i am not sure we need this table in the first place. i thought the info in table `headers` was enough...

**Pull request merged**